src/main/java/org/opentripplanner/middleware/docs/PublicApiDocGenerator.java

```diff
@@ -0,0 +1,375 @@
+package org.opentripplanner.middleware.docs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.opentripplanner.middleware.utils.YamlUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Class that generates the public-facing API documentation.
+ */
+public class PublicApiDocGenerator {
+    private static final Logger LOG = LoggerFactory.getLogger(PublicApiDocGenerator.class);
+    private static final HttpClient client = HttpClient.newBuilder().build();
+    private static final File apiTemplateDefinitionsFile = new File("src/main/resources/api-swagger-template.yaml");
+    private static final String[] PUBLIC_PATHS = new String[] {
```

Comment:
Might be good to give `PUBLIC_PATHS` a comment suggesting that any new public endpoints should be added here.